torch/distributed/optim/zero_redundancy_optimizer.py

```diff
@@ -278,6 +228,22 @@ def partition_parameters(self) -> List[List[Dict]]:
 
         return self._partition_parameters
 
+    @property
+    def index_to_param(self) -> Dict[int, torch.Tensor]:
```

Comment:
these micro-optimizations (cache these calls) mostly make sense for really big models, but since it's the whole point here we use them in fairscale and I figured they would make sense here also